src/main/java/gregtech/common/items/tool/HoeGroundBehavior.java

```diff
@@ -0,0 +1,119 @@
+package gregtech.common.items.tool;
+
+import com.google.common.collect.ImmutableSet;
+import gregtech.api.items.toolitem.ToolHelper;
+import gregtech.api.items.toolitem.aoe.AoESymmetrical;
+import gregtech.api.items.toolitem.behavior.IToolBehavior;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockDirt;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraftforge.event.ForgeEventFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Set;
+
+public class HoeGroundBehavior implements IToolBehavior {
+
+    @Nonnull
+    @Override
+    public EnumActionResult onItemUse(@Nonnull EntityPlayer player, @Nonnull World world, @Nonnull BlockPos pos, @Nonnull EnumHand hand, @Nonnull EnumFacing facing, float hitX, float hitY, float hitZ) {
+        if (facing == EnumFacing.DOWN) return EnumActionResult.PASS;
+
+        ItemStack stack = player.getHeldItem(hand);
+        AoESymmetrical aoeDefinition = ToolHelper.getAoEDefinition(stack);
+
+        Set<BlockPos> blocks;
+        // only attempt to till if the center block is tillable
```

Comment:
This looks like duplicate of part from `GrassPathBehavior`. If it is and it is possible to extracted it then it would be nice, because fixing same problem on multiple places is not fun.